### PR TITLE
Add RA.lengthLt, RA.lengthGt, RA.lengthLte, RA.lengthGte

### PR DESCRIPTION
**Pull request by Undistraction**
See discussion here: #444

I'll add `RA.isLengthEqualTo` and `RA.isLengthNotEqualTo` in a separate PR once this is merged.

### CODE REVIEW

**Comment by Undistraction** [comment]
@guillaumearm Thanks. 

> All functions implemented with complement return true when R.length computes a NaN

Good catch. We should do what Ramda does which is what the underlying JS does, so for the functions using `complement` we must return false for `undefined`, `null` and `NaN` or anything that isn't a number. I'll address this by not using `complement`.

> This functions works with objects that have the 'length' field

Looking at Ramda's docs for `R.length` (which does support objects) there is no mention of the fact. I think this is pretty edge-case and adding to documentation just confuses matters.

**Comment by Undistraction** [comment]
I've removed usage of `R.complement`. I could extract a common function that accepts a comparator predicate:

```
const isLength = predicate => curry((valueLength, value) =>
  compose(flip(predicate)(valueLength), length)(value)
);
```

This should probably be internal.

**Comment by guillaumearm** [comment]
> This is currently internal, but should we expose this publicly?

Very good idea, I think you should open an issue for this.

**Comment by char0n** [comment]
Just my quick observation before I go deeper. I'd drop the `is` prefix from these functions to be consistent with e.g. `path*` or `prop*` functions. `isLengthLt` -> `lengthLt`.

File: src/index.d.ts [hunk 1]
```diff
@@ -810,6 +809,32 @@ declare namespace RamdaAdjunct {
...
+         * Returns `true` if the supplied list or string has a length less than `valueLength`.
+         */
+        lengthLt<T>(valueLength: number, list: string | T[]): boolean;
+        lengthLt(valueLength: number): <T>(list: string | T[]) => boolean;
+
+        /**
+         * Returns `true` if the supplied list or string has a length less than or equal to
+         * `valueLength`.
+         */
+        isLenghtLte<T>(valueLength: number, list: string | T[]): boolean;
```
Comment (char0n) [inline src/index.d.ts]: Drop `is` prefix here

File: src/index.d.ts [hunk 2]
```diff
@@ -810,6 +809,32 @@ declare namespace RamdaAdjunct {
...
+        isLenghtLte(valueLength: number): <T>(list: string | T[]) => boolean;
```
Comment (char0n) [inline src/index.d.ts]: Drop is prefix here

File: src/index.d.ts [hunk 3]
```diff
@@ -810,6 +809,32 @@ declare namespace RamdaAdjunct {
...
+
+        /**
+         * Returns `true` if the supplied list or string has a length greater than or equal to
+         * `valueLength`.
+         */
+        isLenghtGte<T>(valueLength: number, list: string | T[]): boolean;
```
Comment (char0n) [inline src/index.d.ts]: Drop is prefix here

File: src/index.d.ts [hunk 4]
```diff
@@ -810,6 +809,32 @@ declare namespace RamdaAdjunct {
...
+        isLenghtGte(valueLength: number): <T>(list: string | T[]) => boolean;
```
Comment (char0n) [inline src/index.d.ts]: Drop is prefix here

File: src/internal/compareLength.js [hunk 5]
```diff
@@ -0,0 +1,8 @@
+import { curry, compose, flip, length } from 'ramda';
+
+const compareLength = comparator =>
```
Comment (char0n) [inline src/internal/compareLength.js]: Pls do the changes that were discussed in https://github.com/char0n/ramda-adjunct/issues/527

File: src/lengthGt.js [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+
+/**
+ * Returns `true` if the supplied list or string has a length greater than `valueLength`.
+ *
+ * @func lengthGt
+ * @memberOf RA
+ * @since {@link https://char0n.github.io/ramda-adjunct/2.8.0|v2.8.0}
+ * @category List
+ * @sig Number -> [*] -> Boolean
+ * @param Number valueLength The length of the list or string
```
Comment (char0n) [inline src/lengthGt.js]:
```js
@param {number} ....
```
This also concerns every other jsdoc param that we have in this PR

File: src/lengthGte.js [hunk 7]
```diff
@@ -0,0 +1,26 @@
+import { gte } from 'ramda';
+
+import compareLength from './internal/compareLength';
+
+/**
+ * Returns `true` if the supplied list or string has a length greater than or equal to
+ * `valueLength`.
+ *
+ * @func isLenghtGte
```
Comment (char0n) [inline src/lengthGte.js]: drop `is` prefix

File: src/lengthGte.js [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+ * @since {@link https://char0n.github.io/ramda-adjunct/2.8.0|v2.8.0}
+ * @category List
+ * @sig Number -> [*] -> Boolean
+ * @param Number valueLength The length of the list or string
+ * @param Number value The list or string
+ * @return {boolean}
+ * @see {@link http://ramdajs.com/docs/#gte|gte}, {@link http://ramdajs.com/docs/#length|length}
+ * @example
+ *
+ * RA.isLenghtGte(3, [1,2,3,4]); //=> true
```
Comment (char0n) [inline src/lengthGte.js]: Drop `is` prefix from example

File: src/lengthLt.js [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+
+/**
+ * Returns `true` if the supplied list or string has a length less than `valueLength`.
+ *
+ * @func lengthLt
+ * @memberOf RA
+ * @since {@link https://char0n.github.io/ramda-adjunct/2.8.0|v2.8.0}
+ * @category List
+ * @sig Number -> [*] -> Boolean
+ * @param Number valueLength The length of the list or string
```
Comment (char0n) [inline src/lengthLt.js]: fix param sytnax

File: src/lengthLte.js [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+
+/**
+ * Returns `true` if the supplied list or string has a length less than or equal to `valueLength`.
+ *
+ * @func lengthLte
+ * @memberOf RA
+ * @since {@link https://char0n.github.io/ramda-adjunct/2.8.0|v2.8.0}
+ * @category List
+ * @sig Number -> [*] -> Boolean
+ * @param Number valueLength The length of the list or string
```
Comment (char0n) [inline src/lengthLte.js]: fix param syntax

File: test/lengthGt.js [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+    eq(RA.lengthGt(1, undefined), false);
+    eq(RA.lengthGt(1, null), false);
+    eq(RA.lengthGt(1, {}), false);
+    eq(RA.lengthGt(1, true), false);
+    eq(RA.lengthGt(1, false), false);
+    eq(RA.lengthGt(1, 5), false);
+  });
+
+  it(`should be curried`, function() {
+    eq(RA.lengthGt(1)([1, 2]), true);
```
Comment (char0n) [inline test/lengthGt.js]: Add
```js
eq(RA.lengthGt(1, [1, 2]), true);
```

**Comment by Undistraction** [comment]
@char0n Those should all be addressed.

File: src/lengthGt.js [hunk 12]
```diff
@@ -0,0 +1,24 @@
...
+
+/**
+ * Returns `true` if the supplied list or string has a length greater than `valueLength`.
+ *
+ * @func lengthGt
+ * @memberOf RA
+ * @since {@link https://char0n.github.io/ramda-adjunct/2.8.0|v2.8.0}
+ * @category List
+ * @sig Number -> [*] -> Boolean
+ * @param {Number} valueLength The length of the list or string
```
Comment (char0n) [inline src/lengthGt.js]: Number should be lowercase `number`. It's a primitive not a wrapper object.

**Comment by Undistraction** [comment]
@char0n The codebase is full of inconsistencies on this front (try searching for `@return {Boolean}` for example. I resisted fixing it all in this PR to keep it focussed but I'll try and find time to fix all the other issues in another PR.

**Comment by char0n** [comment]
@Undistraction can we pls rebase this onto master ASAP. Nothing is preventing this for being in next release.

**Comment by Undistraction** [comment]
@char0n Are you happy for me to squash and merge?

**Comment by char0n** [comment]
@Undistraction yep go fo it!

**Pull request merged**